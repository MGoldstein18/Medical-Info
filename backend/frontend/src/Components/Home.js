//import React, CSS, and components to sign in and create the different accounts
//Bootstrap is also used for styling
import React from "react";
import "../App.css";
import CreateLO from "./LovedOne/Createlo.js";
import Createm from "./Medical/createm.js";
import SignIn from "./SignIn.js";

class Home extends React.Component {
  render() {
    return (
      <div className="container-xs mt-5">
        <div id="heading" className="shadow-lg m-5 jumbotron jumbotron-fluid">
          <h1 className="mt-5">AMIT - Save a life</h1>
          <p className="lead m-5">
            Share critical medical information instantly with medical
            professionals using your ID number or Popl
          </p>
          <div className="row">
            <div className="col-lg">
              {" "}
              <button className="shadow-lg p-3 m-5 btn-lg btn-warning">
                <a style={{ color: "#AC2020" }} href="#lovedOne">
                  Create Account
                </a>
              </button>
            </div>
            <div className="col-lg">
              <SignIn />
            </div>
          </div>
        </div>

        <section className="mt-5 intro row">
          <p
            style={{ color: "#F47552" }}
            className="col-lg align-self-start text-center"
          >
            When a paramedic arrives at a call...
          </p>
          <p style={{ color: "#D3380D" }} className="col align-self-end">
            <b>information can save a life!</b>
          </p>
        </section>
        <br/>

        <section className="questions row mt-5">
          <p
            style={{ color: "#E37892" }}
            className="col-lg align-self-start text-center"
          >
            Does the paramedic know if you have <b>allergies?</b>
          </p>
          <p
            style={{ color: "#D9456A" }}
            className="col-lg align-self-center text-center"
          >
            Does your mom have a <b>pre-existing condition?</b>
          </p>
          <p
            style={{ color: "#BA264B" }}
            className="col-lg align-self-end text-center"
          >
            Is your son on <b>chronic medication?</b>
          </p>
        </section>

        <section className="m-5 p-5 action text-center">
          <p>
            Use <b>AMIT</b> to instantly share critical information with medical
            professionals using your ID number or Popl
          </p>
          <button className="shadow-lg p-3 mb-5 m-5 btn-lg btn-warning">
            <a style={{ color: "#AC2020" }} href="#lovedOne">
              Create Account
            </a>
          </button>
        </section>

        <section className="row">
          <div
            id="lovedOne"
            className="shadow-lg p-3 mb-5 col-lg container p-5 text-center"
          >
            <div className="family row">
              <p className="col align-self-start text-center">Mom & Dad</p>
              <p className="col align-self-end">Brother & Sister</p>
            </div>
            <div className="family row">
              <p className="col align-self-start text-center">Son & Daughter</p>
              <p className="col align-self-end">
                <b>All our loved ones...</b>
              </p>
            </div>
            <CreateLO />

            <p className="mt-3">
              <b>Safe and Confident</b>
            </p>
            <p style={{ fontSize: "22px" }}>
              Medical professionals will have access to information if they need
              to save your life
            </p>

            <p className="mt-5">
              <b>Secure and In Control</b>
            </p>
            <p style={{ fontSize: "22px" }}>
              Control what information is accessible and know who accesses your
              information and when
            </p>
          </div>
          <div
            id="medical"
            className="shadow-lg p-3 mb-5 col-lg container p-5 text-center"
          >
            <div id="profs">
              <p>Paramedics,</p>
              <p>Doctors,</p>
              <p>Medical professionals,</p>
            </div>

            <p id="future" className="text-center p-5">
              <b>Welcome to the future!</b>
            </p>
            <Createm />
            <ul className="text-center medical-list">
              <li> ~ Built for use in the field</li>
              <li> ~ Reliable information</li>
              <li> ~ Efficient Access</li>
            </ul>
          </div>
        </section>
      </div>
    );
  }
}

export default Home;
