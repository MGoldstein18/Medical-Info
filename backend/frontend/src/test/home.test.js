//a snapshot test of the home page - check that it renders correctly
import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Components/Home.js';

it('renders correctly', () => {
  const tree = renderer
    .create(<Home></Home>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
