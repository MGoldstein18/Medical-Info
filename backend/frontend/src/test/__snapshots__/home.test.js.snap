// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container mt-5"
>
  <div
    className="shadow-lg m-5 jumbotron"
    id="heading"
  >
    <h1
      className="mt-5"
    >
      AMIT - Save a life
    </h1>
    <p
      className="lead m-5"
    >
      Share critical medical information instantly with medical professionals using your ID number or Popl
    </p>
    <button
      className="shadow-lg p-3 mb-5 m-5 btn-lg btn-warning"
    >
      <a
        href="#lovedOne"
        style={
          Object {
            "color": "#AC2020",
          }
        }
      >
        Create Account
      </a>
    </button>
    <button
      className="shadow-lg p-3 mb-5 m-5 btn-lg btn btn-warning"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "color": "#AC2020",
        }
      }
      type="button"
    >
      Sign In
    </button>
  </div>
  <section
    className="mt-5 intro row"
  >
    <p
      className="col align-self-start text-center"
      style={
        Object {
          "color": "#F47552",
        }
      }
    >
      When a paramedic arrives at a call
    </p>
    <p
      className="col align-self-end"
      style={
        Object {
          "color": "#D3380D",
        }
      }
    >
      <b>
        information can save a life!
      </b>
    </p>
  </section>
  <section
    className="questions row mt-5"
  >
    <p
      className="col align-self-start text-center"
      style={
        Object {
          "color": "#E37892",
        }
      }
    >
      Does the paramedic know if you have 
      <b>
        allergies?
      </b>
    </p>
    <p
      className="col align-self-center text-center"
      style={
        Object {
          "color": "#D9456A",
        }
      }
    >
      Does your mom have a 
      <b>
        pre-existing condition?
      </b>
    </p>
    <p
      className="col align-self-end text-center"
      style={
        Object {
          "color": "#BA264B",
        }
      }
    >
      Is your son on 
      <b>
        chronic medication?
      </b>
    </p>
  </section>
  <section
    className="m-5 p-5 action text-center"
  >
    <p>
      Use 
      <b>
        AMIT
      </b>
       to instantly share critical information with medical professionals using your ID number or Popl
    </p>
    <button
      className="shadow-lg p-3 mb-5 m-5 btn-lg btn-warning"
    >
      <a
        href="#lovedOne"
        style={
          Object {
            "color": "#AC2020",
          }
        }
      >
        Create Account
      </a>
    </button>
  </section>
  <section
    className="row"
  >
    <div
      className="shadow-lg p-3 mb-5 mr-5 col container p-5 text-center"
      id="lovedOne"
    >
      <div
        className="family row"
      >
        <p
          className="col align-self-start text-center"
        >
          Mom & Dad
        </p>
        <p
          className="col align-self-end"
        >
          Brother & Sister
        </p>
      </div>
      <div
        className="family row"
      >
        <p
          className="col align-self-start text-center"
        >
          Son & Daughter
        </p>
        <p
          className="col align-self-end"
        >
          <b>
            All our loved ones...
          </b>
        </p>
      </div>
      <button
        className="shadow-lg p-3 mb-5 m-5 btn-lg btn btn-danger"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "color": "#FCBA36",
          }
        }
        type="button"
      >
        <b>
          Create a 
        </b>
        <i>
          Loved One
        </i>
         
        <b>
          Account
        </b>
      </button>
      <p
        className="mt-3"
      >
        <b>
          Safe and Confident
        </b>
      </p>
      <p
        style={
          Object {
            "fontSize": "22px",
          }
        }
      >
        Medical professionals will have access to information if they need to save your life
      </p>
      <p
        className="mt-5"
      >
        <b>
          Secure and In Control
        </b>
      </p>
      <p
        style={
          Object {
            "fontSize": "22px",
          }
        }
      >
        Control what information is accessible and know who accesses your information and when
      </p>
    </div>
    <div
      className="shadow-lg p-3 mb-5 col ml-5 container p-5 text-center"
      id="medical"
    >
      <div
        id="profs"
      >
        <p>
          Paramedics,
        </p>
        <p>
          Doctors,
        </p>
        <p>
          Medical professionals,
        </p>
      </div>
      <p
        className="text-center p-5"
        id="future"
      >
        <b>
          Welcome to the future!
        </b>
      </p>
      <button
        className="shadow-lg p-3 mb-5 m-5 btn-lg btn btn-info"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "color": "#003049",
          }
        }
        type="button"
      >
        <b>
          Create a 
        </b>
        <i>
          Medical
        </i>
         
        <b>
          Account
        </b>
      </button>
      <ul
        className="text-center medical-list"
      >
        <li>
           ~ Built for use in the field
        </li>
        <li>
           ~ Reliable information
        </li>
        <li>
           ~ Efficient Access
        </li>
      </ul>
    </div>
  </section>
</div>
`;
